docs/manual/scala/guide/broker/MessageBrokerApi.md

```diff
@@ -56,6 +62,20 @@ Your broker implementation may provide additional metadata with messages which y
 
 The [`messageKeyAsString`](api/com/lightbend/lagom/scaladsl/api/broker/Message.html#messageKeyAsString:String) method is provided as a convenience for accessing the message key. Other properties can be accessed using the [`get`](api/com/lightbend/lagom/scaladsl/api/broker/Message.html#get\(com.lightbend.lagom.scaladsl.api.broker.MetadataKey[Metadata]\):Metadata) method. A full list of the metadata keys available for Kafka can be found [here](api/com/lightbend/lagom/scaladsl/broker/kafka/KafkaMetadataKeys$.html).
 
+### Skipping messages
+
+You may only want to apply your logic to a subset of the messages that the topic publishes and skip the others. The `Flow` that is passed to [`Subscriber.atLeastOnce`](api/com/lightbend/lagom/scaladsl/api/broker/Subscriber.html#atLeastOnce\(flow:akka.stream.scaladsl.Flow[Payload,akka.Done,_]\):scala.concurrent.Future[akka.Done]) must emit exactly one `Done` element for each element that it receives. It must also emit them in the same order that the elements were received. This means that you must not use methods such as `filter` or `collect` on the `Flow` which would drop elements.
+
+The easiest way to achieve this is to use a total function which returns `Done` for the elements that should be skipped. For example:
+
+@[subscribe-to-topic-skip-messages](code/docs/scaladsl/mb/AnotherServiceImpl.scala)
+
+If your logic is more complicated then you may want to create a separate `Flow` for it. You will also need another `Flow` which maps the elements you want to skip to `Done`, in a similar way to the previous example but as its own stream. Once you have the separate streams you can use `Partition` to split the stream and `Merge` to join them back together. The challenge here is keeping the `Done` elements from each of the separate streams in order. An example of this is:
+
+@[subscribe-to-topic-partition-messages](code/docs/scaladsl/mb/AnotherServiceImpl.scala)
+
+Using this approach, your `logic` stream must request exactly one element from upstream for each request from downstream. If this stream were to request more elements then they could end up going through the `ignore` stream and overtake an element from the `logic` stream.
```

Comment:
Thus this should be checked by the implementation, not rely on luck that people respect this :)